log-collector/src/test/scala/io/phdata/pulse/logcollector/LogCollectorRoutesTest.scala

```diff
@@ -154,4 +155,16 @@ class LogCollectorRoutesTest
       responseAs[String] == "Log Collector"
     }
   }
+
+  test("return runtimeException on bad request to kudu") {
+    when(kuduService.save(Matchers.any(), Matchers.any()))
+      .thenThrow(new RuntimeException("Save failed"))
+
+    val caught = intercept[RuntimeException] {
+      kuduService.save(Matchers.any(), Matchers.any())
```

Comment:
In this test we want to test that the _controller_ is returning a 500 when an exception happens, so the call inside the intercept block should be a call to the controller. All tests in this class are tests of the controller, not the service.
- set up the mock exception (done)
- call the controller 
- assert that a 500 error is returned